apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,80 @@
+import { IPreferenceChannels, StepTypeEnum } from '@novu/shared';
+import { DiscoverWorkflowOutputPreferences } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      workflowId: template.name,
+      tags: template.tags,
+      active: template.active,
+      payload: template.data ? { schema: template.data } : undefined, // Assuming data is mapped to payload schema
+      controls: template.payloadSchema ? { schema: template.payloadSchema } : undefined,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
+      code: template.name, // Assuming name is used as code
+      steps: template.steps.map(WorkflowTemplateGetMapper.toStepDto),
+      name: template.name,
+      description: template.description,
+      origin: template.origin,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+    };
+  }
+  static toMinifiedWorkflowDto(template: NotificationTemplateEntity): MinifiedResponseWorkflowDto {
+    return {
+      _id: template._id,
+      workflowId: template.name,
+      tags: template.tags,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+      stepSummery: template.steps.map(WorkflowTemplateGetMapper.buildStepSummery),
+    };
+  }
+
+  static toWorkflowsMinifiedDtos(templates: NotificationTemplateEntity[]): MinifiedResponseWorkflowDto[] {
+    return templates.map(WorkflowTemplateGetMapper.toMinifiedWorkflowDto);
+  }
+  static toStepDto(step: NotificationStepEntity): StepDto {
+    return {
+      code: step.name || 'Missing Name', // Assuming name is used as code
+      stepId: step._id || step.uuid || 'Missing-ID', // Assuming _id is used as stepId
+      type: step.template?.type || StepTypeEnum.EMAIL, // Assuming type is directly available; adjust as necessary
```

Comment:
i those cases we should return undefined, and let the client deal with the copy.